src/js/components/FormField/FormField.js

```diff
@@ -451,7 +459,44 @@ const FormField = forwardRef(
         {...outerProps}
         style={outerStyle}
         onFocus={(event) => {
-          setFocus(shouldKeepFocus());
+          setFocus(containsFocus(formFieldRef.current));
+          const formDescendants =
+            formFieldRef.current.getElementsByTagName('*');
+          const focusableDescendants = [];
+          for (let i = 0; i < formDescendants.length; i += 1) {
+            const child = formDescendants[i];
+            if (isFocusable(child)) focusableDescendants.push(child);
+
+            let isGrommetInputComponent = false;
+            if (
+              (children &&
+                children.type &&
+                grommetInputNames.includes(children.type.displayName)) ||
+              (!children &&
+                component &&
+                grommetInputNames.includes(component.displayName)) ||
+              (!children && !component)
+            )
+              isGrommetInputComponent = true;
+
+            /* If the FormField has multiple focusable descendants and it 
+            is a grommet input component we should remove the focus only 
+            when one of the focusable descendants contains focus that is 
+            not the first focusable descendant.
+
+            If the FormField child is not a grommet input component we 
+            should remove focus anytime we have a focusable descendant 
+            that has focus. */
+            if (
+              (focusableDescendants.length >= 2 &&
+                isGrommetInputComponent &&
+                child === document.activeElement) ||
+              (focusableDescendants.length >= 1 &&
+                !isGrommetInputComponent &&
+                focusableDescendants.includes(document.activeElement))
+            )
+              setFocus(false);
```

Comment:
I think you can actually remove all this if you change `containsFocus` like noted below. I tested the result and it seems to work well